### ISSUE TITLE
Throw errors for non-class decorators

### ISSUE THREAD

**Issue by ikatyang** [typescript-eslint/typescript-eslint#185]
**What code were you trying to parse?**

```ts
@dec()
const a = 1
```

**What did you expect to happen?**

Throw an error.

**What actually happened?**

No `decorators` attached nor errors were thrown.

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.1.0` |
| `TypeScript`                           | `3.2.2` |
| `node`                                 | `X.Y.Z` |
| `npm`                                  | `X.Y.Z` |

---

Original post: prettier/prettier#5823

**Comment by armano2** [comment]
we are reporting error for decorators: 
```
case 1206: // ts 3.2 "Decorators are not valid here."
```
you should enable `errorOnTypeScriptSyntacticAndSemanticIssues` option

**Comment by JamesHenry** [comment]
We need to be careful recommending that to Prettier at this stage @armano2 - there’s a major difference in performance to think about right now.

**Comment by JamesHenry** [comment]
We should measure for the Prettier case specifically, there’s a big user base to consider

**Comment by JamesHenry** [comment]
I'm working on this now

**Comment by JamesHenry** [comment]
We already have precedent for adding decorators on node's which don't allow them, so I am thinking we should just do that here too to solve them problem for now.

Enabling the diagnostics is expensive, it looks like it's around 30% slower for real world usage of Prettier.

That might be fine in mine cases, might not in others, hard to tell. But given this specific issue, and the fact we already have precedent for this, I think handling it specifically is probably the best way to go.

**Comment by JamesHenry** [comment]
In the future, it would be really nice if there were a way to cheaply report diagnostics on a single `SourceFile`, rather than needing a `Program` cc @typescript-eslint/typescript-team

**Comment by JamesHenry** [comment]
I'll open a PR shortly

**Comment by uniqueiniquity** [comment]
RE: the diagnostics on a single sourcefile, they're technically there, but not exposed in the API. I'll ask around tomorrow to see if that's a safe thing to use/depend on or not
